statsmodels/discrete/discrete_model.py

```diff
@@ -1173,6 +1174,280 @@ def hessian(self, params):
+        p = self.parameterization
+        exog = self.exog
+        endog = self.endog[:,None]
+        mu = self.predict(params)[:,None]
+        mu_p = np.power(mu, p)
+        mu_y = mu_p * endog
+        alphamu = 1 + alpha * mu_p
+        alphamu_y = mu + (alphamu - 1) * endog
+        dalpha = (mu_y * ((endog - 1) / alphamu_y - 1 / alphamu) - 
+                  mu_y / alphamu - alphamu_y * mu_p / (alphamu **2 ))
+        palphamu = alpha * p * mu ** (p - 1)
+        palphamu_y = palphamu * endog
+        dparams = exog * (1 / mu + (endog - 1) * (1 + palphamu_y) / alphamu_y -
+                   (endog * palphamu / alphamu) -
+                   (1 + palphamu_y) / alphamu + alphamu_y * palphamu / 
+                   (alphamu * alphamu))
+        return np.r_[dparams.sum(0), dalpha.sum()]
+
+    def hessian(self, params):
+        """
+        Generalized Poisson model Hessian matrix of the loglikelihood
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+
+        Returns
+        -------
+        hess : ndarray, (k_vars, k_vars)
+            The Hessian, second derivative of loglikelihood function,
+            evaluated at `params`
+
+        Notes
+        -----
+        """
+        return approx_hess(params, self.loglike)
+
+        if self._transparams: # got lnalpha during fit
+            alpha = np.exp(params[-1])
+        else:
+            alpha = params[-1]
+
+        params = params[:-1]
+        p = self.parameterization
+        exog = self.exog
+        endog = self.endog[:,None]
+        mu = self.predict(params)[:,None]
+        mu_p = np.power(mu, p)
+        mu_pm1 = mu_p / mu
+        mumu_p = mu_p * mu_p
+        mu_y = mu_p * endog
+        alphamu = 1 + alpha * mu_p
+        alphamu_y = mu + (alphamu - 1) * endog
+        palphamu = alpha * p * mu ** (p - 1)
+        palphamu_y = palphamu * endog
+
+        # for dl/dparams dparams
+        dim = exog.shape[1]
+        hess_arr = np.empty((dim+1,dim+1))
+
+        for i in range(dim):
+            for j in range(dim):
+                if j > i:
+                    continue
+                hess_arr[i,j] = np.sum(0, axis=0) # need to implement
+        tri_idx = np.triu_indices(dim, k=1)
+        hess_arr[tri_idx] = hess_arr.T[tri_idx]
+
+        # for dl/dparams dalpha
+        dldpda = np.sum(exog * ((endog - 1) * (alphamu_y * palphamu / alpha -
+                        (1 + palphamu_y) * endog * mu_p) / (alphamu_y ** 2) -
+                        ((2 * palphamu_y * (alphamu / alpha - mu_p) / alpha -
+                        mu_p) / (alphamu ** 2)) + 0.), axis=0)
+
+        hess_arr[-1,:-1] = dldpda
+        hess_arr[:-1,-1] = dldpda
+
+        # for dl/dalpha dalpha
+        dldada = (endog * endog * mumu_p * (1 - endog) / (alphamu_y *
+                  alphamu_y) + (mumu_p * (endog + 1) - 2 * mu_p *
+                  (mu_p * mu + endog * mumu_p) / alphamu) /
+                  (alphamu * alphamu))
```

Comment:
Expressions both here and in `score` can be simplified quite a bit with the substitution `a1 = alpha * mu**p`.  I worked through some analogous algebra for the NegativeBinomial case in #3592 (look at individual commits for bite-size algebraic steps).

Sorry to harp on the naming thing, but a big part of the reason why short variable names are problematic is because they make it hard to search in a text editor (or grep) and immediately see everywhere the variable is used without getting a ton of false-positives.  The alphamu, alphamu_y, palphamu, palphamu_y... causes similar trouble.